storage/buckets.rb

```diff
@@ -199,6 +199,133 @@ def delete_bucket project_id:, bucket_name:
   # [END delete_bucket]
 end
 
+def set_retention_policy project_id:, bucket_name:, retention_period:
+  # [START storage_set_retention_policy]
+  # project_id       = "Your Google Cloud project ID"
+  # bucket_name      = "Name of your Google Cloud Storage bucket"
+  # retention_period = "Object retention period defined in seconds"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project_id: project_id
+  bucket  = storage.bucket bucket_name
+
+  bucket.retention_period = retention_period
+
+  puts "Retention period for #{bucket_name} is now #{retention_period} seconds"
```

Comment:
I would have more confidence in this output if it accessed the bucket fields: `#{bucket.name} is now #{bucket.retention_period}`